### PR TITLE
CI: Add build on VFX Reference Platform 2023

### PR DESCRIPTION
**Pull request by garyo**
See #124

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -43,6 +43,20 @@ jobs:
             cxx-compiler: clang++
             cc-compiler: clang
             compiler-desc: Clang
+          - name_prefix: Linux CentOS 7 VFX CY2023
+            release_prefix: linux-vfx2023
+            ostype: linux
+            distro: centos7
```
Comment (jfpanisset) [inline .github/workflows/build.yml:49]: The 2023 aswf-docker containers are based on Rocky Linux 8 (8.8 as of the time they were published). Looks like the only reference to the `distro` symbol is to check whether it is ubuntu or not, which really seems to mean whether the build is in a `aswf-docker` container or not. Changing this value to `rocky8` would work, but perhaps the symbol should be called `aswfdockerbuild` and be a boolean?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -43,6 +43,20 @@ jobs:
             cxx-compiler: clang++
             cc-compiler: clang
             compiler-desc: Clang
+          - name_prefix: Linux CentOS 7 VFX CY2023
```
Comment (jfpanisset) [inline .github/workflows/build.yml:46]: Name should probably be `Linux Rocky 8 VFX CY2023`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -43,6 +43,20 @@ jobs:
...
+            release_prefix: linux-vfx2023
+            ostype: linux
+            distro: centos7
+            os: ubuntu-latest
```
Comment (jfpanisset) [inline .github/workflows/build.yml:50]: This is also present for the 2021 and 2022 aswf-docker containers, and seems to get written to a `OSNAME` env var.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -43,6 +43,20 @@ jobs:
...
+            container: aswf/ci-base:2023
+            vfx-cy: 2023
+            has_cmake_presets: false
+            buildtype: Release
+            conan_version: 2.0.4
```
Comment (jfpanisset) [inline .github/workflows/build.yml:55]: `aswf-docker` containers already have a version of Conan, currently 1.58.0, but it probably makes sense to use your own version across all builds, especially given the flux created by the transition from 1.x to 2.x

**Review by jfpanisset — APPROVED** [review]
Should be good to go in as is to test builds, and address comments later if relevant.

File: .github/workflows/build.yml (hunk 1 above)
Comment (garyo) [reply .github/workflows/build.yml]: Fixed.

File: .github/workflows/build.yml (hunk 3 above)
Comment (garyo) [reply .github/workflows/build.yml:50]: Not sure what fix this comment is suggesting. I'm not sure OSNAME is used, but not worth removing at the moment. The `os` itself is used to select the OS to run the CI build on.

Comment (jfpanisset) [reply .github/workflows/build.yml:50]: I think this comment is obsolete, your second commit got rid of the confusing `distro: centos7` label which was confusing me with `os: ubuntu-latest` but of course the `os:` label is for the runner itself, and the `distro:` was meant for what's inside the container running the build.

**Pull request merged**